index.src.html

```diff
@@ -276,32 +285,58 @@ <h4 id="shortlinks">Navigation</h4>
 <section>
   <h2 id="framework">Framework</h2>
 
-  An IPv4 address is a <dfn export local-lt="private">private address</dfn> if
-  it matches the <a>private address space</a> defined in Section 3 of
-  [[!RFC1918]], a <dfn export local-lt="local">local address</dfn> if it matches
-  the "loopback" space (`127.0.0.0/8`) defined in section 3.2.1.3 of
-  [[!RFC1122]] or the "link-local" space (`169.254.0.0/16`) defined in
-  [[!RFC3927]], and a <dfn export local-lt="public">public address</dfn>
-  otherwise.
-
-  An IPv6 address is a <a>local address</a> if it matches the "Unique Local
-  Address" prefix (`fc00::/7`) defined in Section 3 of [[!RFC4193]] or the
-  "link-local" prefix (`fe80::/10`) defined in section 2.5.6 of [[!RFC4291]],
-  and a <a>public address</a> otherwise.
-
-  ISSUE(mikewest/cors-rfc1918#3): What does "private" mean in the context of
-  IPv6?
+  <h3 id="address-space-heading">Address Space</h3>
+
+  Every IP address belongs to an IP <dfn export>address space</dfn>, which can
+  be one of three different values:
+
+  1.  <dfn export>local</dfn>: contains the local host only. In other words,
+      addresses whose target differs for every device.
+  1.  <dfn>private</dfn>: contains addresses that have meaning only within the
+      current network. In other words, addresses whose target differs based on
+      network position.
+  1.  <dfn>public</dfn>: contains all other addresses. In other words, addresses
+      whose target is the same for all devices globally on the IP network.
+
+  The <a>address space</a> of a given IP address (|address|) is determined
+  in accordance with the IANA Special-Purpose Address Registries
+  ([[IPV4-REGISTRY]] and [[IPV6-REGISTRY]]):
+
+  1.  If |address| is an IPv4 address in the "Loopback" address block
+      (`127.0.0.1/8` at time of writing), then its <a>address space</a> is
+      <a>local</a>.
+  1.  If |address| is an IPv6 address in the "Loopback Address" address block
+      (`::1/128` at time of writing), then its <a>address space</a> is
+      <a>local</a>.
+  1.  If |address| is an IPv6 address in the "IPv4-mapped Address" address block
+      (`::ffff:0:0/96` at time of writing), then its <a>address space</a> is
+      that of its embedded IPv4 address.
+  1.  If |address| belongs to an address block for which the
+      `Globally Reachable` is set to `False` in the relevant IANA registry, then
+      its <a>address space</a> is <a>private</a>.
+  1.  Otherwise the <a>address space</a> of |address| is <a>public</a>.
+
+  For convenience, we additionally define the following terms:
+
+  1.  A <dfn export>local address</dfn> is an address whose <a>address space</a>
+      is <a>local</a>.
+  1.  A <dfn export>private address</dfn> is an address whose
+      <a>address space</a> is <a>private</a>.
+  1.  A <dfn export>public address</dfn> is an address whose
+      <a>address space</a> is <a>public</a>.
```

Comment:
Since you're defining these here, it might be better not to export the `local`/`private`/`public` values.

Reply:
Good point. Removed the `export` attribute on these, since they are mostly syntactic sugar to avoid having to say "an IP address in the XXX address space" every time in this document.